infra/Pos/Communication/Relay/Types.hs

```diff
@@ -44,10 +45,17 @@ instance Buildable PropagationMsg where
     build (DataOnlyPM conts) =
         Buildable.build conts
 
--- | Queue of InvMsges which should be propagated.
-type RelayPropagationQueue = TBQueue (Microsecond, PropagationMsg)
+data RelayContext m = RelayContext
```

Comment:
> And you also need some data to implement this interface. AFAIU from Pos.Launcher.Resource the only necessary data is DiscoveryContextSum.

This is not true. Prior to this patch, we also needed the relay queue `TBQueue`. With this patch, we need the abstract relaying mechanism granted by time-warp. Fact is, the data required to implement the interface may also use effects from the application monad, which is not necessarily `IO` nor `MonadIO m => m`. To claim that all we need in order to implement relaying, always and forever, is a `DiscoveryContextSum`, is to leave relaying unacceptably concrete.